management_api_app/service_bus/deployment_status_update.py

```diff
@@ -56,7 +56,7 @@ async def receive_message():
                         await receiver.complete_message(msg)
 
 
-def create_updated_deployment_document(workspace: Workspace, message: DeploymentStatusUpdateMessage):
+def create_updated_deployment_document(workspace: dict, message: DeploymentStatusUpdateMessage):
```

Comment:
why are we changing to a dict here? why not a resource? -- this used to return a workspace - but now it returns a dict and the callers expect a workspace/resource type -- the class should also have a return type

Reply:
this is due to the fact that now we get deployment status update both for workspace and workspace service (and soon user resource) those objects have different properties. And to reuse the existing  code I had to make it generic.

Reply:
per our talk earlier we okd the dict - but should it not be resource then instead of workspace?